cli/src/commands/parse.rs

```diff
@@ -98,8 +103,8 @@ pub fn parse_lockfile(
         }
     }
 
-    // If the path is neither a valid manifest nor lockfile, we abort.
-    if !parser.is_path_manifest(&path) {
+    // Abort if generation is disabled or path is neither lockfile nor manifest.
+    if !generate_lockfiles || !parser.is_path_manifest(&path) {
```

Comment:
There is a problem with the logic in this block. The `unreachable` macro is triggered with the changes from this PR. Here is an example from [a library project](https://github.com/deepcausality-rs/deep_causality) where there are only manifests:

```
../deep_causality  9  19 on  main [?] via 🦀 v1.73.0
❯ /Users/maxrake/dev/phylum/localdev/cli/target/debug/phylum --version
phylum v5.8.1-4-ge70c5c1

../deep_causality  9  19 on  main [?] via 🦀 v1.73.0
❯ /Users/maxrake/dev/phylum/localdev/cli/target/debug/phylum parse --no-generation ./Cargo.toml
thread 'main' panicked at cli/src/commands/parse.rs:112:21:
internal error: entered unreachable code: neither lockfile nor manifest
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Great catch, thanks.